CRM/MembershipExtras/Form/PaymentScheme.php

```diff
@@ -151,8 +164,54 @@ private function addPaymentProcessorField() {
     }
     $select = ['' => ts('- select -')] + $paymentProcessors;
 
-    $this->add('select', 'payment_processor', E::ts('Payment Processor'), $select, TRUE);
+    $this->add('select', 'payment_processor', ts('Payment Processor'), $select, TRUE);
 
   }
 
+  private function isDeleteContext() {
+    return $this->_action == CRM_Core_Action::DELETE;
+  }
+
+  private function postProcessDeletion() {
+    $contributionRecursCount = \Civi\Api4\ContributionRecur::get()
+      ->selectRowCount()
+      ->addWhere('payment_plan_extra_attributes.payment_scheme_id', '=', $this->id)
+      ->execute()->count();
+
+    if ($contributionRecursCount > 0) {
+      CRM_Core_Session::setStatus(ts('You are not able to delete this payment scheme as it’s linked with one or more recurring contributions.'), ts("Payment Scheme cannot be deleted"), "error");
```

Comment:
And when I move this  method to the BAO class , I would throw an exception in the place of this, and here in the form class I would catch the exception and do the `CRM_Core_Session::setStatus` if there is an error